controllers/core/openstackversion_controller.go

```diff
@@ -0,0 +1,295 @@
+	if instance.Status.DeployedVersion != nil && instance.Spec.TargetVersion != *instance.Status.DeployedVersion {
+		cl = append(cl, *condition.UnknownCondition(corev1beta1.OpenStackVersionMinorUpdateOVNControlplane, condition.InitReason, string(corev1beta1.OpenStackVersionMinorUpdateInitMessage)),
+			*condition.UnknownCondition(corev1beta1.OpenStackVersionMinorUpdateControlplane, condition.InitReason, string(corev1beta1.OpenStackVersionMinorUpdateInitMessage)))
+		// fixme add dataplane conditions here
+		//condition.UnknownCondition(corev1beta1.OpenStackVersionMinorUpdateOVNDataplane, condition.InitReason, string(corev1beta1.OpenStackVersionMinorUpdateInitMessage)),
+		//condition.UnknownCondition(corev1beta1.OpenStackVersionMinorUpdateDataplane, condition.InitReason, string(corev1beta1.OpenStackVersionMinorUpdateInitMessage)))
+	}
+	instance.Status.Conditions.Init(&cl)
+	if isNewInstance {
+		// Register overall status immediately to have an early feedback e.g. in the cli
+		return ctrl.Result{}, nil
+	}
+
+	instance.Status.Conditions.Set(condition.FalseCondition(
+		corev1beta1.OpenStackVersionInitialized,
+		condition.RequestedReason,
+		condition.SeverityInfo,
+		corev1beta1.OpenStackVersionInitializedReadyRunningMessage))
+
+	instance.Status.AvailableVersion = &envAvailableVersion
+	defaults := openstack.InitializeOpenStackVersionImageDefaults(ctx, envContainerImages)
+	// store the defaults for the currently available version
+	if instance.Status.ContainerImageVersionDefaults == nil {
+		instance.Status.ContainerImageVersionDefaults = make(map[string]*corev1beta1.ContainerDefaults)
+	}
+	instance.Status.ContainerImageVersionDefaults[envAvailableVersion] = defaults
+
+	// calculate the container images for the target version
+	val, ok := instance.Status.ContainerImageVersionDefaults[instance.Spec.TargetVersion]
+	if !ok {
+		Log.Info("Target version not found in defaults", "targetVersion", instance.Spec.TargetVersion)
+		return ctrl.Result{}, nil
+	}
+	instance.Status.ContainerImages = openstack.GetContainerImages(ctx, val, *instance)
+
+	instance.Status.Conditions.MarkTrue(
+		corev1beta1.OpenStackVersionInitialized,
+		corev1beta1.OpenStackVersionInitializedReadyMessage)
+	Log.Info("OpenStackVersion Initialized")
+
+	// lookup the current Controlplane object
+	controlPlane := &corev1beta1.OpenStackControlPlane{}
+	err = r.Client.Get(ctx, client.ObjectKey{
+		Namespace: instance.Namespace,
+		Name:      instance.Name,
+	}, controlPlane)
+	if err != nil {
+		// we ignore not found
+		if k8s_errors.IsNotFound(err) {
+			Log.Info("No controlplane found.")
+			return ctrl.Result{}, nil
+		} else {
+			return ctrl.Result{}, err
+		}
+	}
+
+	// greenfield deployment //FIXME check dataplane here too
+	if controlPlane.Status.DeployedVersion == nil {
+		Log.Info("No controlplane or controlplane is not deployed")
+		return ctrl.Result{}, nil
+	}
+
+	// TODO minor update for OVN Dataplane in progress
```

Comment:
Will this come in a follow up PR or we hold this PR until the dataplane implementation is ready?

Reply:
same as above. There is no way to implement this in this patch (and keep minor updates passing) until we land the appropriate Dataplane code. If I were to wire this in now, minor updates would fail to work...

Reply:
OK

Reply:
fixed in the lastest push